### DISCUSSION TITLE
How to combine states of generated providers?

### DISCUSSION THREAD

**Discussion by abigotado** [rrousselGit/riverpod#1981] [Q&A] [answered]
Hello!

Is it possible to combine states of generated providers?

For example, if we take an official example for combining providers:
```dart
final weatherProvider = FutureProvider((ref) async {
  // We use `ref.watch` to listen to another provider, and we pass it the provider
  // that we want to consume. Here: cityProvider
  final city = ref.watch(cityProvider);

  // We can then use the result to do something based on the value of `cityProvider`.
  return fetchWeather(city: city);
});
```
and refactor it for code-generation, it can be smth like this:
```dart
@riverpod
weatherProvider(ref) async {
  // We use `ref.watch` to listen to another provider, and we pass it the provider
  // that we want to consume. Here: cityProvider
  final city = ref.watch(cityProvider);

  // We can then use the result to do something based on the value of `cityProvider`.
  return fetchWeather(city: city);
}
```
Is that a right way?

And, well, what to do with Notifier/AsyncNotifier? If we need to refresh AsyncNotifierProvider depending on the state of some other provider, how can we achieve that?

**Comment by kennethjyang** [comment, marked as answer]
Here's the documentation for how to combine states using the new code generation syntax: https://docs-v2.riverpod.dev/docs/concepts/combining_providers.

And here's another discussion that talks about how `ref` is available already inside the new (Async)NotifierProviders so you can just use the other notifiers: #1964
